server/stream.go

```diff
@@ -3024,10 +3112,15 @@ func (mset *stream) startingSequenceForSources() {
 		if ssi.iname == _EMPTY_ {
 			ssi.setIndexName()
 		}
-		si := &sourceInfo{name: ssi.Name, iname: ssi.iname, sf: ssi.FilterSubject}
+		si := &sourceInfo{name: ssi.Name, iname: ssi.iname, sf: ssi.FilterSubject, sfs: ssi.FilterSubjects}
 		// Check for transform.
 		if ssi.SubjectTransformDest != _EMPTY_ {
 			si.tr, _ = NewSubjectTransform(ssi.FilterSubject, ssi.SubjectTransformDest) // can not return an error because validated in AddStream
+		} else if len(ssi.SubjectTransformDests) != 0 {
+			si.trs = make([]*subjectTransform, len(ssi.SubjectTransformDests))
```

Comment:
Is there a reason to have both `si.tr` and `si.trs`? It seems like both use-cases can be satisfied by just having the latter (and a non-plural config option would just end up in an array length of 1).

Reply:
My concern here is that by having effectively two codepaths in each of these places, there are all sorts of existing tests that aren't covering this new code. Ideally we would consolidate paths down as much as possible.

Reply:
Totally understand. I added (modified) some of the existing tests to use multiple filters. But I am not a big fan of having to have 2 code paths with copied but slightly modified code in both places, and I would rather just have the array. But if I change it then it means we forgo a bit of backwards compatibility. i.e. an 'old' application would make a request with just the single fields but would get returned those fields empty and the array fields filled with one entry instead (or if they do a StreamInfo later). Are we ok with that?

Reply:
Indeed.. for backwards compat `StreamSource` and `StreamSourceInfo` need to retain the existing singular fields, specifically for embedded usage. However, the internal code path for applying any logic could normalize to the slice.

Reply:
My point is if you normalize to an array then if you have an array length of 1 you can't tell whether that was initially set in the array field or was initially set in the singular field.

Reply:
For the purpose of differentiating on the `StreamSourceInfo`?

Reply:
Yes: does an application putting a single filter in 'filterSubject' also expect to see a single filter in 'filterSubject' in the StreamInfo?